### PR TITLE
Expose the peer that propagates a message to the recipient

### PR DESCRIPTION
**Pull request by aschmahmann**
This is a solution for #215. We were doing a lot of passing around `(src, Message)` to avoid exposing the source next to the message, so I hope that solving #215 is a good thing 😬.

I haven't removed all the places where we pass both `src` and `Message`, but got most of the internal ones.

### CODE REVIEW

File: go.mod
Comment (raulk) [inline go.mod:15]: Accidental version bump.

File: pubsub.go
Comment (raulk) [inline pubsub.go]: nit: just `id` is fine. `hostID` sounds alien, as hosts don't have IDs, but rather peers do.

**Review by raulk — APPROVED** [review]
This looks like a pretty good rationalisation to me anyway, even if we didn't have to address #218.

File: go.mod
Comment (aschmahmann) [reply go.mod:15]: What's the deal with the go version being in the go.mod? Any reason we should remove it? GoLand keeps adding it for me automatically whenever I compile.

Comment (raulk) [reply go.mod:15]: The go1.13 toolchain keeps doing this. On re-reading the [modules docs released with go1.12](https://golang.org/doc/go1.12#modules), I think it's fine to keep this directive.

I thought it specified the minimum go toolchain version that's needed to build this module, and that builds with earlier versions (e.g. go1.12) would simply fail. That's not the case, though. Builds will succeed unless they use language features available in go1.13 that aren't available in the toolchain in use. When that happens, this directive allows the compiler to give better hints when barking.

Comment (raulk) [reply go.mod:15]: Conclusion: ok to keep this.

**Comment by raulk** [comment]
Let's wait for @vyzo to sign-off too.

**Review by vyzo — APPROVED** [review]
LGTM

**Comment by vyzo** [comment]
Test failure is systemic, @aschmahmann you need to `go fmt`!

```
*** go fmt ./...

ERROR: some files not gofmt'ed:

./floodsub_test.go
```

**Comment by aschmahmann** [comment]
> @aschmahmann you need to go fmt!

lol that's what I get for trying to rename a variable using the Github text editor

**Pull request merged**
